youtube/feed.go

```diff
@@ -298,97 +359,45 @@ func (p *Plugin) parseYtUrl(channelUrl *url.URL) (idType ytUrlType, id string, e
 	// Prefix check allows method to provide a more helpful error message,
 	// when attempting to parse an invalid handle URL.
 	if strings.HasPrefix(path, "@") {
-		handle := ytHandleRegex.FindStringSubmatch(path)
-		if handle != nil {
-			return ytUrlTypeHandle, handle[1], nil
+		if ytHandleRegex.MatchString(path) {
+			return &searchChannelID{ id: path }, nil
 		} else {
-			return ytUrlTypeInvalid, id, fmt.Errorf("%#v is not a valid youtube handle", path)
+			return nil, fmt.Errorf("\"%s\" is not a valid youtube handle", path)
 		}
 	}
 
-	pathSegments := strings.Split(channelUrl.Path, "/")
-	if len(pathSegments) != 3 {
-		return ytUrlTypeInvalid, id, fmt.Errorf("%s is not a valid path", path)
+	pathSegments := strings.Split(path, "/")
+	if len(pathSegments) != 2 {
+		return nil, fmt.Errorf("\"%s\" is not a valid path", path)
 	}
 
-	first := pathSegments[1]
-	second := pathSegments[2]
+	first := pathSegments[0]
+	second := pathSegments[1]
 
 	switch first {
 	case "shorts":
 		return p.parseYtVideoID(second)
 	case "channel":
-		id = ytChannelIDRegex.FindString(second)
-		if id != "" {
-			return ytUrlTypeChannel, id, nil
+		if ytChannelIDRegex.MatchString(second) {
+			return &channelID{ id: second }, nil
 		} else {
-			return ytUrlTypeInvalid, id, fmt.Errorf("%s is not a valid youtube channel id", id)
+			return nil, fmt.Errorf("\"%s\" is not a valid youtube channel id", id)
```

Comment:
```suggestion
			return nil, fmt.Errorf("%q is not a valid youtube channel id", id)
```

Reply:
Swapping `%s` for `%q` results characters like newlines being printed as their character literals, such as `\n`...

If this is preferable to using `\"%s\"`, I'll change those.

Reply:
Can the input ever contain newlines? Regardless, I'd argue such a scenario is an extreme edge case at best; further, even if we stick with the original formatting, the result will be strange -- consider
```
"foo
bar" is not a valid youtube channel id
```
In light of that, I feel `%q` is probably the way to go.

Reply:
Ok, absolutely. Generally, I think displaying the escape sequences for characters in messages to end users, isn't a great idea, and Des changed them to `%s` when doing #1468, but I'd agree, it would make the formatting weird.